src/graphql/account.rs

```diff
@@ -418,6 +418,60 @@ impl AccountMutation {
         update_jwt_expires_in(expires_in)?;
         Ok(time)
     }
+
+    /// Retrieves the user's language selection by username.
+    #[graphql(guard = "RoleGuard::new(super::Role::SystemAdministrator)
+        .or(RoleGuard::new(super::Role::SecurityAdministrator))
+        .or(RoleGuard::new(super::Role::SecurityManager))
+        .or(RoleGuard::new(super::Role::SecurityMonitor))")]
+    async fn language(&self, ctx: &Context<'_>, username: String) -> Result<Option<String>> {
+        let store = crate::graphql::get_store(ctx).await?;
+        let map = store.account_map();
+
+        let account = map
+            .get(&username)?
+            .ok_or_else::<async_graphql::Error, _>(|| "Account not found".into())?;
+
+        Ok(account.language)
+    }
+
+    /// Updates only the user's language setting, regardless of the user's role.
+    #[graphql(guard = "RoleGuard::new(super::Role::SystemAdministrator)
+        .or(RoleGuard::new(super::Role::SecurityAdministrator))
+        .or(RoleGuard::new(super::Role::SecurityManager))
+        .or(RoleGuard::new(super::Role::SecurityMonitor))")]
+    async fn update_language(
+        &self,
+        ctx: &Context<'_>,
+        username: String,
+        language: Option<UpdateLanguage>,
```

Comment:
Could you explain why you used `Option` for `UpdateLanguage`?

This API is for changing the language, so why do we include a case where users aren't trying to change it?

Reply:
Besides, the fields of `UpdateLanguage`, `old` and `new` are all `Option`s. I think this includes cases we don't need to consider:  `Some`(`None`, `None`) and `None`

Reply:
I realize it was a mistake to use an `Option` as the argument type for the function. I've since revised the code to simplify it by excluding unnecessary considerations. I would appreciate it if you could review the changes and let me know if there's anything that could be improved.